app/controllers/form_items_controller.rb

```diff
@@ -0,0 +1,21 @@
+class FormItemsController < ApplicationController
+  # authorization via cancan
+  load_and_authorize_resource
+
+  def update
+    attrs = params[:form_item]
+
+    if attrs[:ancestry].blank? or !@form_item.check_ancestry_integrity(attrs[:ancestry])
+      return render nothing: true, status: 422
+    end
+
+    @form_item.ancestry = attrs[:ancestry]
```

Comment:
It will be easier on the frontend if we just accept parent_id instead of ancestry. You should then be able to just do `@form_item.parent = FormItem.find(params[:parent_id])`. Ancestry will handle rest.